csv-adapter/src/main/java/ch/unibas/dmi/dbis/polyphenydb/adapter/csv/CsvStore.java

```diff
@@ -103,4 +111,41 @@ public void updateColumnType( Context context, CatalogColumn catalogColumn ) {
     public String getAdapterName() {
         return ADAPTER_NAME;
     }
+
+
+    @Override
+    public List<AdapterSetting> getAvailableSettings() {
+        return AVAILABLE_SETTINGS;
+    }
+
+
+    @Override
+    public void updateSettings( Map<String, String> newSettings ) {
+        for ( AdapterSetting s : AVAILABLE_SETTINGS ) {
+            if ( newSettings.containsKey( s.name ) ) {
+                if ( s.modifiable ) {
+                    String newValue = newSettings.get( s.name );
+                    if ( !s.canBeNull && newValue == null ) {
+                        throw new RuntimeException( "Setting \"" + s.name + "\" is not allowed to be null!" );
+                    }
+                    if ( newValue.equals( settings.get( s.name ) ) ) {
+                        //noinspection SwitchStatementWithTooFewBranches
+                        switch ( s.name ) {
+                            case "directory":
+                                csvDir = new File( newValue );
+                                break;
+
+                            default:
+                                throw new RuntimeException( "Missing entry for setting \"" + s.name + "\"!" );
+                        }
+                        settings.put( s.name, newValue );
+                    } else {
+                        // Same value, do nothing
+                    }
+                } else {
+                    throw new RuntimeException( "Setting \"" + s.name + "\" is not modifiable!" );
+                }
+            }
+        }
+    }
```

Comment:
A lot of this seems to be code that should be extracted into a more generic form. Afaict the only part that will differ between stores is the `switch` statement in the middle (L133 - L140).

So we could probably move the outer part into abstract `Store` class, and then have a private abstract function `applySetting`?

Reply:
You're absolutely right. This is definitely not a nice solution. Thanks for pointing out. It is now as you suggested.